source/custom/nixnet_service.custom.cpp

```diff
@@ -1,2 +1,292 @@
+            }
+            case nixnet_grpc::ReadStateRequest::StateIdEnumCase::STATE_ID_ENUM_NOT_SET: {
+                return ::grpc::Status(::grpc::INVALID_ARGUMENT, "The value for state_id was not specified or out of range");
+                break;
+            }
+        }
+
+        u32 state_size = GetStateSize(state_id);
+        // Since GetStateSize returns only u32, in case of error, we are returning a meaningful message
+        if (state_size == ::grpc::INVALID_ARGUMENT)
+        {
+            return ::grpc::Status(::grpc::INVALID_ARGUMENT, "The value for state_id was not specified or out of range");
+        }
+        response->mutable_state_value()->mutable_raw_value()->resize(state_size,0);
+        nxStatus_t fault {};
+
+        auto status = library_->ReadState(session_ref, state_id, state_size, const_cast<char*>(response->mutable_state_value()->mutable_raw_value()->data()), &fault);
+
+        response->set_status(status);
+        if (status == 0) {
+            void* state_value_raw = (void*)response->state_value().raw_value().data();
+            switch(state_id){
+                case nixnet_grpc::ReadState::READ_STATE_STATE_TIME_CURRENT :{
+                    response->mutable_state_value()->set_time_current(*(nxTimestamp100ns_t*)state_value_raw);
+                    break;
+                } 
+                case nixnet_grpc::ReadState::READ_STATE_STATE_TIME_COMMUNICATING :{
+                    response->mutable_state_value()->set_time_communicating(*(nxTimestamp100ns_t*)state_value_raw);
+                    break;
+                } 
+                case nixnet_grpc::ReadState::READ_STATE_STATE_TIME_START :{
+                    response->mutable_state_value()->set_time_start(*(nxTimestamp100ns_t*)state_value_raw);
+                    break;
+                }
+                case nixnet_grpc::ReadState::READ_STATE_STATE_CAN_COMM :{
+                    SetCanComm(*(u32*)state_value_raw , response->mutable_state_value()->mutable_can_comm());
+                    break;
+                }
+                case nixnet_grpc::ReadState::READ_STATE_STATE_FLEX_RAY_COMM :{
+                    SetFlexRayComm(*(u32*)state_value_raw , response->mutable_state_value()->mutable_flex_ray_comm());
+                    break;
+                }
+                case nixnet_grpc::ReadState::READ_STATE_STATE_LIN_COMM :{
+                    SetLinComm((u32*)state_value_raw, response->mutable_state_value()->mutable_lin_comm());
+                    break;
+                }
+                case nixnet_grpc::ReadState::READ_STATE_STATE_SESSION_INFO :{
+                    response->mutable_state_value()->set_session_info(*(u32*)state_value_raw);
+                    break;
+                }
+                case nixnet_grpc::ReadState::READ_STATE_STATE_FLEX_RAY_STATS :{
+                    convert_to_grpc(*(_nxFlexRayStats_t*)state_value_raw, response->mutable_state_value()->mutable_flex_ray_stat());
```

Comment:
Field name should be changed to `flex_ray_stats`

Reply:
Done